### PR TITLE
ospf: Various fixes including much improved assert

### PR DESCRIPTION
**Pull request by ddutt**
This includes a much improved assert that addresses a bunch of issues in the existing model. It also includes a bug fix for summarize.

Test updates to follow soon

### CODE REVIEW

File: suzieq/engines/pandas/ospf.py [hunk 1]
```diff
@@ -280,41 +290,42 @@ def aver(self, **kwargs):
...
+            .query('~adjState.isin(["full", "passive"]) and '
+                   'ifState == "down"') \
+            .reset_index(drop=True)
+        ifdown_df['assertReason'] += \
+            ifdown_df.ifState.apply(lambda x: ['ifdown'])
+
+        # Validate loopback interfaces are of the right network type
+        passed_df['assertReason'] += passed_df.apply(
+            lambda x: ['network type !loopback']
+            if (any(ele in x.ifname for ele in ['lo', 'oopbac']) and
```
Comment (LucaNicosia) [inline suzieq/engines/pandas/ospf.py:310]: is it right the array `['lo', 'oopbac']` or it is a typo on the second element?

File: suzieq/engines/pandas/ospf.py [hunk 2]
```diff
@@ -280,41 +290,42 @@ def aver(self, **kwargs):
...
 
         # In case of Junos, the OSPF interface name is a subif of
         # the interface name. So, create a new column with the
         # orignal interface name.
-        ospf_df['lldpIfname'] = ospf_df['ifname'].str.split('.').str[0]
-        int_df = ospf_df[use_cols].merge(
+        failed_df['lldpIfname'] = failed_df['ifname'].str.split('.').str[0]
+        use_cols = failed_df.columns
+
+        int_df = failed_df[use_cols].merge(
```
Comment (LucaNicosia) [inline suzieq/engines/pandas/ospf.py:328]: If use_cols == failed_df.columns, I don't think you need to do `failed_df[use_cols]` right?

File: suzieq/engines/pandas/ospf.py (hunk 1 above)
Comment (ddutt) [reply suzieq/engines/pandas/ospf.py:310]: Yes, some NOS have lo as their loopback name, others use Loopback, and some other loopback

File: suzieq/engines/pandas/ospf.py (hunk 2 above)
Comment (ddutt) [reply suzieq/engines/pandas/ospf.py:328]: True enough

File: suzieq/engines/pandas/ospf.py [hunk 3]
```diff
@@ -323,106 +334,170 @@ def aver(self, **kwargs):
...
+            dflist: List[pd.DataFrame]
+            result: str: user specification of what result to return
+
+        Returns:
+            The assert result dataframe
+        """
+        ospf_df = pd.concat(dflist)
+        ospf_df['result'] = ospf_df.assertReason.apply(
+            lambda x: 'fail' if x else 'pass')
+        # Weed out the loopback, SVI interfaces as they have no LLDP peers
```
Comment (claudious96) [inline suzieq/engines/pandas/ospf.py:495]: I think this is a leftover comment after a refactoring, right?

**Pull request merged**